src/modules-lua/noit/module/googleanalytics4.lua

```diff
@@ -0,0 +1,565 @@
+                'dfpViewableImpressionsPercent',
+                'dfpClicks',
+                'dfpCTR',
+                'dfpRevenue',
+                'dfpRevenuePer1000Sessions',
+                'dfpECPM'
+            }
+        }
+    elseif params.metric_group == 'DoubleClickForPublishersBackfill' then
+        -- Restricted metrics - can only be queried under certain circumstances,
+        -- unable to verify data received during testing...
+        params.metrics = {
+            {
+                'backfillImpressions',
+                'backfillCoverage',
+                'backfillMonetizedPageviews',
+                'backfillImpressionsPerSession',
+                'backfillViewableImpressionsPercent',
+                'backfillClicks',
+                'backfillCTR',
+                'backfillRevenue',
+                'backfillRevenuePer1000Sessions',
+                'backfillECPM'
+            }
+        }
+    elseif params.metric_group == 'DoubleClickBidManager' then
+        -- Restricted metrics - can only be queried under certain circumstances,
+        -- unable to verify data received during testing...
+        params.metrics = {
+            {
+                'dbmCPA',
+                'dbmCPC',
+                'dbmCPM',
+                'dbmCTR',
+                'dbmClicks',
+                'dbmConversions',
+                'dbmCost',
+                'dbmImpressions',
+                'dbmROAS'
+            }
+        }
+    elseif params.metric_group == 'DoubleClickSearch' then
+        -- Restricted metrics - can only be queried under certain circumstances,
+        -- unable to verify data received during testing...
+        params.metrics = {
+            {
+                'dsCPC',
+                'dsCTR',
+                'dsClicks',
+                'dsCost',
+                'dsImpressions',
+                'dsProfit',
+                'dsReturnOnAdSpend',
+                'dsRevenuePerClick'
+            }
+        }
+    else
+        local err_msg = 'unknown googleanalytics4 metric group (' .. params.metric_group .. ')'
+        mtev.log('error', '%s\n', err_msg)
+        check.bad()
+        check.unavailable()
+        check.status(err_msg)
+        return
+    end
+
+    -- assume the worst.
+    check.bad()
+    check.unavailable()
+
+    local metric_count = 0
+    local check_metric = function (name,value)
+        if value == nil then
+            check.metric_double(name, value)
+        elseif tonumber(value) ~= nil then
+            if string.find(value, "%.") ~= nil then
+                check.metric_double(name, value)
+            else
+                check.metric_uint64(name, value)
+            end
+        else
+            check.metric_string(name, value)
+        end
+        metric_count = metric_count + 1
+    end
+
+    -- Check about every 15 minutes
+    if (current_time - cache_table[uuid]['timestamp']) >= (current_time - (60 * 15)) then
```

Comment:
Isn't the `cache_table[uuid]['timestamp']` the number of seconds since epoch?

If so, you'd want current_time only on one side.

I would think something like:

` if ( cache_table[uuid]['timestamp'] + (60 * 15) >= current_time ) then `